src/Microsoft.ML.Data/Training/TrainerEstimatorBase.cs

```diff
@@ -104,16 +102,15 @@ private void CheckInputSchema(SchemaShape inputSchema)
             // may define their own requirements on the label column.
             if (LabelColumn != null)
             {
-                var labelCol = inputSchema.FindColumn(LabelColumn.Name);
-                if (labelCol == null)
+                if (!inputSchema.TryFindColumn(LabelColumn.Name, out var labelCol))
                     throw Host.Except($"Label column '{LabelColumn.Name}' is not found");
                 CheckLabelCompatible(labelCol);
             }
         }
 
         protected virtual void CheckLabelCompatible(SchemaShape.Column labelCol)
         {
-            Contracts.AssertValue(labelCol);
+            Contracts.CheckValue(labelCol, nameof(labelCol));
```

Comment:
>CheckValue [](start = 22, length = 10)

actually the existence is checked by the base class that's calling this method. Why `Check` ?

Reply:
It's a bit of a pedantic point, and perhaps we ought to be more systematic than I was here, but the point is that it is entirely possible for user code to trigger an assertion (e.g., silly person subclasses and immediately calls this method), which goes against all our existing guidelines for asserts and contracts.

Nonetheless I think it's generally a good policy to follow... the distinction between assert telling me "this should never happen" and check telling me "this can happen but if it does I'm going to complain" is one I find incredibly useful, when attempting to comprehend all this code we've built.

---
In reply to: [215723997](https://github.com/dotnet/machinelearning/pull/845#discussion_r215723997) [](ancestors = 215723997)